Jenkinsfile

```diff
@@ -0,0 +1,319 @@
+               destTag = 'promoteTest'
+               serviceName = 'maingateway-service'
+           }
+           when {
+               expression {
+                   env.userInput == 'Gateway' || env.userInput == 'All'
+               }
+           }
+           steps {
+               echo "Deploy to ${TEST_PROJECT} "
+               promoteServiceSetup(params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, 'maingateway-service',params.IMAGE_REGISTRY, params.IMAGENAMESPACE, env.destTag, params.TEST_PROJECT)    
+               promoteService(params.IMAGENAMESPACE, params.TEST_PROJECT,'maingateway-service', env.srcTag, env.destTag)
+           }
+        }
+        stage('Pushing to Test - fisuser') {
+            environment {
+                srcTag = 'latest'
+                destTag = 'promoteTest'
+                serviceName = 'fisuser-service'
+            }
+           when {
+               expression {
+                   env.userInput == 'FisUser' || env.userInput == 'All'
+               }
+           }
+            steps {
+                echo "Deploy to ${TEST_PROJECT} "
+                promoteServiceSetup(params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, 'fisuser-service',params.IMAGE_REGISTRY, params.IMAGENAMESPACE, env.destTag, params.TEST_PROJECT)    
+                promoteService(params.IMAGENAMESPACE, params.TEST_PROJECT, 'fisuser-service', env.srcTag, env.destTag)
+            }
+        }
+        stage('Pushing to Test - fisalert') {
+            environment {
+                srcTag = 'latest'
+                destTag = 'promoteTest'
+                serviceName = 'fisalert-service'
+            }
+           when {
+               expression {
+                   env.userInput == 'FisAlert' || env.userInput == 'All'
+               }
+           }
+            steps {
+                echo "Deploy to ${TEST_PROJECT} "
+                promoteServiceSetup(params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, 'fisalert-service',params.IMAGE_REGISTRY, params.IMAGENAMESPACE, env.destTag, params.TEST_PROJECT)
+                promoteService(params.IMAGENAMESPACE, params.TEST_PROJECT, 'fisalert-service', env.srcTag, env.destTag)
+            }
+        }
+        stage('Pushing to Test - nodejsalert') {
+            environment {
+                srcTag = 'latest'
+                destTag = 'promoteTest'
+                serviceName = 'nodejsalert-service'
+            }
+           when {
+               expression {
+                   env.userInput == 'UI' || env.userInput == 'All'
+               }
+           }
+            steps {
+                echo "Deploy to ${TEST_PROJECT} "
+                promoteServiceSetup(params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, 'nodejsalert-ui',params.IMAGE_REGISTRY, params.IMAGENAMESPACE, env.destTag, params.TEST_PROJECT)
+                promoteService(params.IMAGENAMESPACE, params.TEST_PROJECT, 'nodejsalert-ui', env.srcTag, env.destTag)
+            }
+        }
```

Comment:
I think we should have one stage for 'Integration Tests'  here before moving to production env.

Reply:
And for the test, we don't even have any testing code available.
We don't need another test environment at least for now.

Reply:
It is not adding an Integration test stage could make the deployment status available.
A testing (Junit/API integration test) module is missing and need to be developed.